tests/integration/watch-del-all.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/* Copyright 2015-present Facebook, Inc.
+ * Licensed under the Apache License, Version 2.0 */
+
+class watchDelAllTestCase extends WatchmanTestCase {
+  function testWatchDelAll() {
+    $dir = PhutilDirectoryFixture::newEmptyFixture();
+    $root = realpath($dir->getPath());
+
+    $files = array('a', 'b', 'c', 'd', 'e');
+    $dirs = array_map(
+      function ($file) use($root) {
+        return "$root/$file";
+      }, 
+      $files
+    );
+
+    foreach ($dirs as $d) {
+      mkdir($d);
+
+      foreach ($files as $f) {
+        touch("$d/$f");
+      }
+
+      $this->watch($d);
+      $this->assertFileList($d, $files); 
+    }
+
+    $resp = $this->watchmanCommand('watch-list');
+    $watched = $resp['roots'];
+    sort($watched);
+
+    $this->assertEqual($dirs, $watched);
+
+    $this->watchmanCommand('watch-del-all');
+
+    $this->waitForWatchman(
```

Comment:
I'd expect this to be an empty list immediately because the implementation is removing from the watch list immediately; I think you can implement this as a simple assertEquals on the result `watchmanCommand`?